bin/ttab

```diff
@@ -156,22 +157,30 @@ case $terminalApp in
     # Note: 'iTerm.app' is what $TERM_PROGRAM contains when running from iTerm.app
     iTerm=1
     ;;
+  'gnome-terminal')
+    gnomeTerminal=1
+    ;;
   *)
     # If an unknown terminal is specified, we issue a warning and fall back to Terminal.app
-    echo "WARNING: '$terminalApp' is not a supported terminal application; defaulting to Terminal.app." >&2    
+    echo "WARNING: '$terminalApp' is not a supported terminal application; defaulting to Terminal.app." >&2
     ;;
 esac
 shopt -u nocasematch
 
-# Set target-terminal-app-appropriate variables used later. 
+# Set target-terminal-app-appropriate variables used later.
 if (( iTerm )); then
   terminalApp='iTerm' # will be used with `activate application`
   # Note: iTerm2's AppleScript syntax changed fundamentally in v3 (for the better, but incompatibly so),
-  #       so we need to distinguish versions below. 
+  #       so we need to distinguish versions below.
   #       $iTermOld reflects a pre-v3 version.
   [[ $(osascript -e 'version of application "iTerm"') =~ ^(1|2) ]] && iTermOld=1 || iTermOld=0
-else # Terminal.app
-  terminalApp='Terminal' # will be used with `activate application`
+else
+  if [ -n "$COLORTERM" ]; then
```

Comment:
On a general note: please use Bash syntax (`[[ ... ]]`) consistently.

I suggest using `GNOME_TERMINAL_SCREEN` or `GNOME_TERMINAL_SERVICE` rather than `COLORTERM` for identifying a Gnome Terminal.

Also, we need to make sure that `$terminalApp` is set both when running from Gnome Terminal and when invoked from a different application via `-a gnome-terminal`.

Therefore, something like:

```suggestion
elif [[ $gnomeTerminal == 1 || -n $GNOME_TERMINAL_SCREEN ]]; then
    terminalApp='gnome-terminal'
    gnomeTerminal=1
else
    terminalApp='Terminal' # will be used with `activate application`
fi
```

Reply:
Applied change.